tests/fixture_valid.js

```diff
@@ -246,14 +177,165 @@ module.exports.checkFixture = function checkFixture(fixture, usedShortNames=[])
+                  description: `channel '${switchedChannel}' is referenced from capability '${cap.name}' (#${i}) in channel '${ch}' but is not defined.`,
+                  error: null
+                });
+              }
+              usedChannels.push(switchedChannel);
+            }
+          }
+          else if (switchesCount > 0) {
+            result.errors.push({
+              description: `Channel '${ch}' uses the 'switchesChannels' property, but its capability '${cap.name}' (#${i}) is missing the 'switchToChannels' property.`,
+              error: null
+            });
+          }
         }
       }
+      else if (switchesCount > 0) {
+        result.errors.push({
+          description: `Channel '${ch}' uses the 'switchesChannels' property, but is missing capabilities (which define to which channels the switching channels should be switched).`,
+          error: null
+        });
+      }
+    }
+
+    for (let i = 1; i <= fixture.modes.length; i++) {
+      const mode = fixture.modes[i-1];
+
+      const modeShortName = mode.shortName || mode.name;
+      if (usedModeShortNames.includes(modeShortName)) {
+        result.errors.push({
+          description: `shortName '${modeShortName}' not unique in mode #${i}.`,
+          error: null
+        });
+      }
+      usedModeShortNames.push(modeShortName);
+
+      if (/\bmode\b/i.test(mode.name) || /\bmode\b/i.test(mode.shortName)) {
+        result.errors.push({
+          description: `mode name and shortName must not contain the word 'mode' in mode '${modeShortName}' (#${i}).`,
+          error: null
+        });
+      }
+
+      if ('physical' in mode
+        && 'lens' in mode.physical
+        && 'degreesMinMax' in mode.physical.lens
+        && mode.physical.lens.degreesMinMax[0] > mode.physical.lens.degreesMinMax[1]
+        ) {
+        result.errors.push({
+          description: `physical.lens.degreesMinMax is an invalid range in mode '${modeShortName}' (#${i}).`,
+          error: null
+        });
+      }
+
+      for (let chIndex = 1; chIndex <= mode.channels.length; chIndex++) {
+        const ch = mode.channels[chIndex-1];
+
+        if (ch === null) {
+          continue;
+        }
+
+        usedChannels.push(ch);
+
+        // it can be a normal channel
+        if (ch in fixture.availableChannels) {
+          if (fixture.availableChannels[ch].type === 'Pan') {
+            checkPanTiltMaxExistence(result, fixture, mode, ch, 'panMax');
+          }
+          else if (fixture.availableChannels[ch].type === 'Tilt') {
+            checkPanTiltMaxExistence(result, fixture, mode, ch, 'tiltMax');
+          }
+          continue;
+        }
+
+        // or it is a fine channel
+        if (ch in fineChannels) {
+          if (!mode.channels.includes(fineChannels[ch])) {
+            result.errors.push({
+              description: `mode '${modeShortName}' uses switching channel '${ch}' (#${chIndex}) but is missing its trigger channel '${fineChannels[ch]}'`,
```

Comment:
Currently, the test only makes sure that the coarse counterpart of a fine channel is used as well. But if a mode uses an 8bit and a 24bit channel, we should ensure that it also contains the 16bit channel, shouldn't we?